src/resumableHelpers.js

```diff
@@ -0,0 +1,96 @@
+// INTERNAL HELPER METHODS (handy, but ultimately not part of uploading)
+export default class ResumableHelpers {
+  /**
+   * Stop the propagation and default behavior of the given event `e`.
+   * @param {Event} e
+   */
+  static stopEvent(e) {
+    e.stopPropagation();
+    e.preventDefault();
+  }
+
+  /**
+   * Generate a unique identifier for the given file based on its size, filename and relative path.
+   * @param {File} file
+   * @returns {string}
+   */
+  static generateUniqueIdentifier(file) {
+    var relativePath = file.webkitRelativePath || file.relativePath || file.fileName || file.name; // Some confusion in different versions of Firefox
+    var size = file.size;
+    return (size + '-' + relativePath.replace(/[^0-9a-zA-Z_-]/img, ''));
+  }
+
+  /**
+   * Flatten the given array and all contained subarrays.
+   * Credit: {@link https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_flattendeep}
+   * @param {*[]} array
+   * @returns {*[]} The flattened array
+   */
+  static flattenDeep(array) {
+    return Array.isArray(array)
+      ? array.reduce((a, b) => a.concat(this.flattenDeep(b)), [])
+      : [array];
+  }
+
+  /**
+   * Filter the given array based on the predicate inside `callback`
+   * and executes `errorCallback` for duplicate elements.
+   * @param {*[]} array
+   * @param {function(*): *} callback
+   * @param {function(*): void} errorCallback
+   */
+  static uniqBy(array, callback, errorCallback) {
+    let seen = new Set();
+    return array.filter((item) => {
+      let k = callback(item);
+      if (seen.has(k)) {
+        errorCallback(item);
+        return false;
+      } else {
+        return seen.add(k);
```

Comment:
I find it quite confusing to `return seen.add(k);`, because `Set.add(k)` returns the set (i.e. an object) after adding `k`.

Since this is inside a `filter` function, this returned set (object) is read as "truthy" and thus everything works, but it would be clearer to the human reader if we wrote this instead:
```suggestion
        seen.add(k);
        return true;
```